### DISCUSSION TITLE
`way_area` in hstore instead of a separate column now?

### DISCUSSION THREAD

**Discussion by iandees** [osm2pgsql-dev/osm2pgsql#1589] [Q&A]
I'm trying to add a feature to Tilezen's vector-datasource, which uses [an osm2pgsql style](https://github.com/tilezen/vector-datasource/blob/master/osm2pgsql.style) file that was created quite a while ago. In there, we have `way_area` defined [here](https://github.com/tilezen/vector-datasource/blob/f5a938edfc6cbe52afcda02bc89a21bb2e3a8c9f/osm2pgsql.style#L22) without any flags specified. Back then, using that style file would add a `way_area` column to the output postgres tables. Nowadays, it seems to put those in the `tags` hstore column. The [osm2pgsql docs](https://osm2pgsql.org/doc/manual.html#special-tags) seem to say that the `way_area` special tag will create a column (not get added to the hstore)

Matching [our vector-datasource docs](https://github.com/tilezen/vector-datasource/wiki/Mapzen-Vector-Tile-Service#load-pbf-data) from back in 2016, I ran:

```
osm2pgsql --slim --hstore-all -C 1024 -S osm2pgsql.style -d tokyoosm /Users/idees/Downloads/kanto-latest.osm.pbf
```

... but this time I'm running it with a more-modern version of osm2pgsql from homebrew:

```
$ osm2pgsql --version
2021-12-08 12:45:27  osm2pgsql version 1.5.1
Build: Release
Compiled using the following library versions:
Libosmium 2.17.0
Proj [API 4] Rel. 7.2.1, January 1st, 2021
Lua 5.1.4 (LuaJIT 2.1.0-beta3)
```

What should I change here to move that way_area entry to a column?

**Comment by joto** [comment]
Between osm2pgsql 1.2.2 and 1.3.0 the behaviour changed. Until 1.2.2 (when used with your options) the `way_area` was **both** in its own column and in the hstore column `tags`. From 1.3.0 on the extra column isn't there any more, just the entry in `tags`. I don't remember any decision to change the somewhat strange behaviour with the two places the `way_area` was written to, so it looks like this came about as part of some cleanup of the code, which just had this side-effect but I can't find anything relevant in the commit messages.

So this is definitely a regression, but I am not sure whether it makes sense to change this now. Is this an actual problem in your case? It should be possible to just change the SQL query accessing this field.

**Reply by iandees** [reply]
Thanks for the context!

Yes, I can probably change the queries to pull from the tags hstore column. I was hoping there was something I could change in the osm2pgsql style to replicate the old behavior. If that's not easy then I can focus on the queries path.

**Reply by pnorman** [reply]
I think maintaining the old behavior is important - way_area as a column is important

**Comment by lonvia** [comment]
The change is that we require a `linear` or `polygon` flag in the style file now for a column to be created. The default style file always had that. Yours doesn't specify any flags for `way_area`. If you add a `linear` in the flag column, all should work as before.
